RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClustersToLegacyAlgo.dev.cc

```diff
@@ -0,0 +1,189 @@
+
+#include <alpaka/alpaka.hpp>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "SiStripClustersToLegacyAlgo.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::sistrip {
+  using namespace ::sistrip;
+  using namespace sistripConverter;
+  using namespace cms::alpakatools;
+
+  class siStripConvKer_fillClCollSlim {
+  public:
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_HOST_ACC void operator()(TAcc const& acc,
```

Comment:
can this be simply
```suggestion
    ALPAKA_FN_HOST_ACC void operator()(Acc1D const& acc,
```
?

Reply:
the compiler complains:
```
src/RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClustersToLegacyAlgo.dev.cc(26): error: calling a __device__ function("auto cms::alpakatools::uniform_elements< ::alpaka::AccGpuUniformCudaHipRt< ::alpaka::ApiCudaRt,     ::std::integral_constant<unsigned long, (unsigned long)1ul> , unsigned int> , unsigned int , void> (const T1 &, T2...)") from a __host__ __device__ function("operator()") is not allowed
        for (auto i : uniform_elements(acc, nbClusterCandidates)) {
                      ^

src/RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClustersToLegacyAlgo.dev.cc(26): error: A non-constexpr __device__ function ("uniform_elements") with "auto" deduced return type cannot be directly referenced outside the bodies of device functions, except if the reference is absent when __CUDA_ARCH__ is undefined
        for (auto i : uniform_elements(acc, nbClusterCandidates)) {
                      ^
```
unless `ALPAKA_FN_HOST_ACC` is changed to compile only device code (`ALPAKA_FN_ACC`). The template usage comes from [this contribution](https://indico.cern.ch/event/1377095/#1-writing-kernels-with-cmssw). It is my understanding that the TAcc is back-end agnostic while the Acc1D is expanded in the concrete accelerator type, hence the reason for the template and host macro too (which in fact is compilation overhead here).

Reply:
Thanks for checking.
Yes, please change to function to `ALPAKA_FN_ACC`: it cannot be used on the host anyway, since it takes an accelerator object as a parameter.